src/cpp/session/SessionSourceDatabaseSupervisor.cpp

```diff
@@ -452,6 +455,20 @@ Error attachToSourceDatabase()
    FilePath existingSdb = sessionDirPath();
    if (existingSdb.exists())
       return sessionDirLock().acquire(sessionLockFilePath(existingSdb));
+   
+   // migrate from 'sdb' to current folder layout if needed
+   bool needsSdbMigration =
+         !sourceDatabaseRoot().exists() &&
+         sdbSourceDatabaseRoot().exists();
+   
+   if (needsSdbMigration)
```

Comment:
The migration codepath should only be hit if the new `sources` folder does not already exist, so in practice this should only occur once.